src/network/network_server.cpp

```diff
@@ -399,8 +399,19 @@ NetworkRecvStatus ServerNetworkGameSocketHandler::SendError(NetworkErrorCode err
 /** Send the check for the NewGRFs. */
 NetworkRecvStatus ServerNetworkGameSocketHandler::SendNewGRFCheck()
 {
+	if (_grfconfig == nullptr) {
+		/* There are no NewGRFs, continue with the game password. */
+		return this->SendNeedGamePassword();
+	}
```

Comment:
If you don't plan to squash this PR, maybe already move this blob to where the next commit will move it to?